### PR TITLE
Add ScoreV2 to mods and change it to bitwise

### PR DESCRIPTION
**Pull request by kyogoi**
Added ScoreV2, changed the mods overall to bitwise computation and added readable comments to further understand dict

### CODE REVIEW

File: discord/bot.rb [hunk 1]
```diff
@@ -84,11 +84,25 @@ def should_add_url(msg) not (OLD_REGEX.match?(msg) and msg.include?('https://osu
...
+    1 << 6  => 'DT',  # 64        - DoubleTime
+    1 << 7  => 'RX',  # 128       - Relax
+    1 << 8  => 'HT',  # 256       - HalfTime
+    1 << 9  => 'NC',  # 512       - Nightcore
+    1 << 10 => 'FL',  # 1024      - Flashlight
+    1 << 11 => 'AT',  # 2048      - Autoplay
+    1 << 12 => 'SO',  # 4096      - SpunOut
+    1 << 13 => 'AP',  # 8192      - AutoPilot, "Relax2"
+    1 << 14 => 'PF',  # 16384     - Perfect
+    1 << 29 => 'SV2', # 536870912 - ScoreV2
```
Comment (christopher-dG) [inline discord/bot.rb]: Change this to `V2` here and in `order`, then realign the comment please

File: discord/bot.rb [hunk 2]
```diff
@@ -84,11 +84,25 @@ def should_add_url(msg) not (OLD_REGEX.match?(msg) and msg.include?('https://osu
 def mods(n)
   return 'None' if n == 0
   mods = {
-    1 => 'NF', 2 => 'EZ', 4 => '', 8 => 'HD', 16 => 'HR', 32 => 'SD', 64 => 'DT',
-    128 => 'RL', 256 => 'HT', 64 | 512 => 'NC', 1024 => 'FL', 2048 => 'AT',
-    4096 => 'SO', 8192 => 'AP', 32 | 16384 => 'PF',
+    1 << 0  => 'NF',  # 1         - NoFail
```
Comment (christopher-dG) [inline discord/bot.rb]: Could we actually change this to `Symbol`s? `'foo'` becomes `:foo`.

File: discord/bot.rb [hunk 3]
```diff
@@ -84,11 +84,25 @@ def should_add_url(msg) not (OLD_REGEX.match?(msg) and msg.include?('https://osu
...
+    1 << 10 => 'FL',  # 1024      - Flashlight
+    1 << 11 => 'AT',  # 2048      - Autoplay
+    1 << 12 => 'SO',  # 4096      - SpunOut
+    1 << 13 => 'AP',  # 8192      - AutoPilot, "Relax2"
+    1 << 14 => 'PF',  # 16384     - Perfect
+    1 << 29 => 'SV2', # 536870912 - ScoreV2
   }
-  order = ['EZ', 'HD', 'HT', 'DT', 'NC', 'HR', 'FL', 'NF', 'SD', 'PF', 'RL', 'SO', 'AP', 'AT']
+
+  order = ['EZ', 'HD', 'HT', 'DT', 'NC', 'HR', 'FL', 'NF', 'SD', 'PF', 'RL', 'SO', 'AP', 'AT', 'SV2']
```
Comment (christopher-dG) [inline discord/bot.rb]: These will also need to be turned to `Symbol`s

File: discord/bot.rb [hunk 4]
```diff
@@ -84,11 +84,25 @@ def should_add_url(msg) not (OLD_REGEX.match?(msg) and msg.include?('https://osu
 def mods(n)
   return 'None' if n == 0
   mods = {
-    1 => 'NF', 2 => 'EZ', 4 => '', 8 => 'HD', 16 => 'HR', 32 => 'SD', 64 => 'DT',
-    128 => 'RL', 256 => 'HT', 64 | 512 => 'NC', 1024 => 'FL', 2048 => 'AT',
-    4096 => 'SO', 8192 => 'AP', 32 | 16384 => 'PF',
+    1 << 0  => 'NF',  # 1         - NoFail
+    1 << 1  => 'EZ',  # 2         - Easy
+    1 << 2 =>  'NV',  # 4         - No Video, deprecated and unused for like 4 years
```
Comment (christopher-dG) [inline discord/bot.rb]: Add one more space before the `=>` to align

**Review by christopher-dG — CHANGES_REQUESTED** [review]
Looks good!

**Comment by kyogoi** [comment]
Changes added

**Comment by christopher-dG** [comment]
Hm, this still breaks (but I don't think it's related to changes you made)

```
irb(main):051:0> mods(32768)
=> "+EZHDHTDTNCHRFLNFSDPFSOAPAT"
```

**Comment by christopher-dG** [comment]
Ah, the broken ones are from mania plays where stuff like `Keys4` isn't handled. I'll take care of that later. Thanks! 😄

**Pull request merged**